Maple2.File.IO/Nif/NifBlock.cs

```diff
@@ -0,0 +1,23 @@
+﻿namespace Maple2.File.IO.Nif;
+
+public class NifBlock {
+    public string BlockType { get; init; }
+    public bool HasName { get; init; }
+    public int BlockIndex { get; init; }
+    public string Name;
+
+    public NifBlock(string blockType, bool hasName, int blockIndex) {
+        BlockType = blockType;
```

Comment:
You can actually just `this.GetType().Name` and not require passing this

Reply:
Wouldn't work with the generic NifBlock placeholders for unimplemented types. Those are passed in from NifDocument too